bgpd/bgp_mplsvpn.c

```diff
@@ -233,6 +236,762 @@ int bgp_nlri_parse_vpn(struct peer *peer, struct attr *attr,
 #undef VPN_PREFIXLEN_MIN_BYTES
 }
 
+static int ecom_intersect(struct ecommunity *e1, struct ecommunity *e2)
```

Comment:
Could this be moved to bgp_ecommunity.c and Could we please add a little description of what the API is trying to do.

Reply:
Using ecom_intersect() is fine but as a second step, we should do some optimization. In EVPN, we have built a hash of VRFs that are interested in a particular RT and so there is just a hash lookup (per RT in route) followed by getting the interested VRFs. @mkanjari or I could help do the optimization too after initial commit of this code.

Reply:
I agree that optimization in this way is a good idea. After this round is committed, let's revisit.

Reply:
sounds fine to me